ext/RMagick/rmutil.c

```diff
@@ -1458,7 +1458,11 @@ rm_progress_monitor(
     VALUE rval;
     VALUE method, offset, span;
 
-    if (ruby_stack_check())
+// Default Ruby minimum stack size
+#define RUBY_VM_THREAD_MACHINE_STACK_SIZE_MIN (  16 * 1024 * sizeof(VALUE)) /*   64 KB or  128 KB */
+
+    // Check stack length manually instead of ruby_stack_check() for old Ruby.
+    if (ruby_stack_length(NULL) > RUBY_VM_THREAD_MACHINE_STACK_SIZE_MIN)
```

Comment:
So do you have any idea why we're going over the stack size?

Reply:
Not yet ... 😣